src/loaders.ts

```diff
@@ -35,26 +35,51 @@ type resolve = (
 	recursiveCall?: boolean,
 ) => MaybePromise<Resolved>;
 
-const extensions = ['.js', '.json', '.ts', '.tsx', '.jsx'] as const;
+const mappedExtensions: Record<string, string[]> = {
+	'.js': ['.ts', '.tsx', 'jsx'],
```

Comment:
It appears to be the other way around:

```suggestion
	'.js': ['.ts', '.tsx'],
	'.jsx': ['.ts', '.tsx'],
```

See this Replit with `tsc`: https://replit.com/@karlhorky/FriendlyWanCharactermapping#index.ts